### PR TITLE
[EOC-420] Add validation to name field when editing cohort/list (front and backend validation)

### PR DESCRIPTION
**Pull request by adamklepacz**

### CODE REVIEW

File: client/src/modules/cohort/components/CohortHeader.jsx [hunk 1]
```diff
@@ -106,7 +122,7 @@ class CohortHeader extends PureComponent {
     } = event;
 
     this.setState({ nameInputValue: value }, () => {
```
Comment (marekrozmus) [inline client/src/modules/cohort/components/CohortHeader.jsx]: not needed braces

File: server/controllers/cohort.js [hunk 2]
```diff
@@ -104,6 +105,12 @@ const updateCohortById = (req, resp) => {
   });
   let cohortActivity;
 
+  if (name !== undefined) {
+    if (!validator.isLength(name, { min: 1, max: 32 })) {
```
Comment (marekrozmus) [inline server/controllers/cohort.js]: one if is enough:
`if (name !== undefined && !validator...)`

File: server/controllers/list.js [hunk 3]
```diff
@@ -418,6 +419,12 @@ const updateListById = (req, resp) => {
   });
   let listActivity;
 
+  if (name !== undefined) {
+    if (!validator.isLength(name, { min: 1, max: 32 })) {
```
Comment (marekrozmus) [inline server/controllers/list.js]: same here

File: client/src/modules/cohort/components/CohortHeader.jsx [hunk 4]
```diff
@@ -171,30 +187,32 @@ class CohortHeader extends PureComponent {
...
     } = this.props;
-    const { nameInputValue } = this.state;
+    const { errorMessageId, nameInputValue } = this.state;
     const nameToUpdate = _trim(nameInputValue);
     const { name: previousName } = details;
 
-    if (_trim(previousName) === nameToUpdate) {
-      this.setState({ isNameInputVisible: false });
+    if (!errorMessageId) {
+      if (_trim(previousName) === nameToUpdate) {
```
Comment (alemikolo) [inline client/src/modules/cohort/components/CohortHeader.jsx]: We get `previousName` from props, from DB. Do we have to trim it? Cohort/list `name` should be saved to DB without trailing spaces so wonder if trimming `previousName` is necessary?

File: client/src/modules/list/components/ListHeader.jsx [hunk 5]
```diff
@@ -138,28 +150,30 @@ class ListHeader extends PureComponent {
...
     } = this.props;
-    const { nameInputValue } = this.state;
+    const { errorMessageId, nameInputValue } = this.state;
     const nameToUpdate = _trim(nameInputValue);
     const { name: previousName } = details;
 
-    if (_trim(previousName) === nameToUpdate) {
-      this.setState({ isNameInputVisible: false });
+    if (!errorMessageId) {
+      if (_trim(previousName) === nameToUpdate) {
```
Comment (alemikolo) [inline client/src/modules/list/components/ListHeader.jsx]: do we have to trim `previousName`?

File: client/src/modules/cohort/components/CohortHeader.jsx (hunk 4 above)
Comment (adamklepacz) [reply client/src/modules/cohort/components/CohortHeader.jsx]: It was here already :)

File: client/src/modules/cohort/components/CohortHeader.jsx [hunk 6]
```diff
@@ -171,30 +185,32 @@ class CohortHeader extends PureComponent {
...
```
Comment (marekrozmus) [inline client/src/modules/cohort/components/CohortHeader.jsx]: why double `if`? why not just:
`if (!errorMessageId && _trim(prev...`?

File: client/src/modules/list/components/ListHeader.jsx [hunk 7]
```diff
@@ -138,28 +148,30 @@ class ListHeader extends PureComponent {
...
```
Comment (marekrozmus) [inline client/src/modules/list/components/ListHeader.jsx]: combine `ifs`

**Pull request merged**